src/master/filesystem.cc

```diff
@@ -218,7 +218,24 @@ int fs_loadall(void) {
 		throw MetadataFsConsistencyException(
 				"temporary metadata file (" + std::string(kMetadataTmpFilename) + ") exists, metadata directory is in dirty state");
 	}
-	if ((metadataserver::isMaster()) && !fs::exists(kMetadataFilename)) {
+	std::string metadataFile = std::string("");
+	if (fs::exists(kMetadataFilename)) {
```

Comment:
This checking even when might look redundant at first, enforce priority for existing actual metadata file over the possibility that there might also be an old legacy metadata file that was left hanging.

Reply:
My point is that first check is not need, because the next checks already guarantee that the right scenario will be handled and the result is the same.

Anyways, the logic will work OK. These are only suggestions :)